proto/spacemesh/v1/tx_types.proto

```diff
@@ -53,3 +53,34 @@ message TransactionState {
     }
     TransactionState state = 2;
 }
+
+// TransactionResultsRequest request object for results stream.
+message TransactionResultsRequest {
+    // id is filter by transaction id.
+    bytes id = 1;
+    // address is a filter by account address, it could be principal or any affected address.
+    bytes address = 2;
+    // start streaming from this layer. if 0 - stream will start from genesis.
+    uint32 start = 3;
+    // end streaming at this layer. if 0 - stream till the latest available layer.
+    uint32 end = 4;
+    // watch live data.
+    bool watch = 5;
+}
+
+message TransactionResult {
+    enum Status {
+        SUCCESS = 0;
+        FAILURE = 1;
+        INVALID = 2;
+    }
+
+    Transaction tx = 1;
+    Status status = 2;
+    string message = 3;
+    uint64 gas_consumed = 4;
+    uint64 fee = 5;
+    bytes block = 6;
+    uint32 layer = 7;
+    repeated bytes updated_addresses = 8;
```

Comment:
I think `touched_addresses` might be a better name. What does this mean, exactly? Just accounts that had their balance updated? What about other state changes?

Reply:
in general this is exactly the list with addresses (accounts) that were updated by this transaction

for example spawn (not self-spawn) will update two addresses:
- principal that pays for gas
- spawned account with additional state (attached template and immutable state)

spend updates 2 accounts:
- principal - gas and balance
- receiver - balance

and so on

Reply:
There are some nuances and challenges here from the VM perspective but we can take this elsewhere, will open an issue to follow up